src/utils/assert.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright 2023 Google LLC.
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export function assert<T>(predicate: T): asserts predicate {
+  if (!predicate) {
+    throw new Error('Internal assertion failed.');
```

Comment:
```suggestion
    throw new UnknownErrorException('Internal assertion failed.');
```

Reply:
Errors get converted to unknown errors. It'd probably be better to keep this as Error since utilities should not depend on other files.